libs/constrained-generators/src/Constrained/Examples/Append.hs

```diff
@@ -0,0 +1,184 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Constrained.Examples.Append where
+
+import Constrained
+import Constrained.Base
+import Constrained.ListSplit
+import Test.QuickCheck (Gen, Property, counterexample, generate, property)
+import Prelude hiding (even)
+
+-- =========================================================
+-- Useful specs to use in Examples
+
+even :: Specification BaseFn Int
+even = MemberSpec [0, 2, 4, 6, 8, 10]
+
+endsWith :: [Int] -> Specification BaseFn [Int]
+endsWith suffix = TypeSpec (ListSpec Nothing [] TrueSpec TrueSpec NoFold (Split Nothing (Just suffix))) []
+
+beginsWith :: [Int] -> Specification BaseFn [Int]
+beginsWith prefix = TypeSpec (ListSpec Nothing [] TrueSpec TrueSpec NoFold (Split (Just prefix) Nothing)) []
+
+-- | Run an example
+go :: forall a. HasSpec BaseFn a => Specification BaseFn a -> IO a
```

Comment:
`go` is a really bad name for a top level function, since it is very common to use this name for internal loops. My suggestion is to actually remove it, since it looks like it is dead code that it is not used anywhere